core/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -172,6 +172,14 @@ private static MapperParsingException wrapInMapperParsingException(SourceToParse
         return new MapperParsingException("failed to parse", e);
     }
 
+    public static String[] splitAndValidatePath(String fullFieldPath) {
+        if (fullFieldPath.contains("..")) {
+            throw new IllegalArgumentException("object field starting or ending with a [.] makes object resolution ambiguous: ["
+                    + fullFieldPath + "]");
+        }
+        return fullFieldPath.split("\\.");
+    }
```

Comment:
what about splitting first and then checking whether the array contains an empty string? I think that would also cover two other cases we are interested in validating: leading dots and trailing dots? (just thinking out loud)